modules/@apostrophecms/image/ui/apos/components/AposMediaUploader.vue

```diff
@@ -0,0 +1,258 @@
+        {
+          'is-dragging': this.dragover
+        }
+      ];
+    }
+  },
+  methods: {
+    dragHandler (event) {
+      event.preventDefault();
+      this.dragging = true;
+    },
+    async uploadMedia (event) {
+      // Set `dragover` in case the media was dropped.
+      this.dragover = false;
+
+      this.$emit('upload-started');
+      const files = event.dataTransfer ? event.dataTransfer.files : event.target.files;
+      const fileCount = files.length;
+
+      const emptyDoc = await apos.http.post(this.action, {
+        body: {
+          _newInstance: true
+        }
+      });
+      await apos.notify(
+        // TODO: i18n
+        `Uploading ${fileCount} image${fileCount > 1 ? 's' : ''}`, {
+          dismiss: true
+        }
+      );
+
+      // Send up placeholders
+      for (const file of files) {
+        this.createPlaceholder(file);
+      }
+
+      const imageIds = [];
+      // Actually upload the images and send them up once all done.
+      for (const file of files) {
+        const img = await this.insertImage(file, emptyDoc);
+        imageIds.push(img._id);
+      }
+
+      // TODO: i18n
+      await apos.notify('Upload Successful', {
+        type: 'success',
+        dismiss: true
+      });
+
+      // When complete, refresh the image grid, with the new images at top.
+      this.$emit('upload-complete', imageIds);
+
+      // TODO: Else if uploading multiple images, show them as a set of selected images for editing.
+    },
+    createPlaceholder(file) {
+      const img = new Image();
+      const dimensions = {};
+      const objectUrl = window.URL.createObjectURL(file);
+      const self = this;
+
+      img.onload = function () {
+        dimensions.width = this.width;
+        dimensions.height = this.height;
+        window.URL.revokeObjectURL(objectUrl);
+
+        self.$emit('create-placeholder', dimensions);
+      };
+      img.src = objectUrl;
+    },
+    async insertImage(file, emptyDoc) {
+      const formData = new window.FormData();
+
+      formData.append('file', file);
+      let attachment;
+
+      // Make an async request to upload the image.
+      try {
+        attachment = await apos.http.post('/api/v1/@apostrophecms/attachment/upload', {
+          body: formData
+        });
+      } catch (error) {
+        console.error('Error uploading media.', error);
+
+        const msg = error.body && error.body.message ? error.body.message : 'Upload error';
+
+        await apos.notify(msg, {
+          type: 'danger',
+          icon: 'alert-circle-icon',
+          dismiss: true
+        });
+        return;
+      }
+
+      const imageData = Object.assign(emptyDoc, {
+        title: attachment.title,
+        attachment
+      });
+
+      try {
+        const imgPiece = await apos.http.post(this.action, {
+          body: imageData
+        });
+
+        return imgPiece;
+      } catch (error) {
+        await this.notifyErrors(error, 'Upload Error');
```

Comment:
I think this method doesn't exist in the uploader component, it was left behind in the parent component.